logic/src/commonMain/kotlin/com/wire/kalium/logic/data/message/receipt/DetailedReceipt.kt

```diff
@@ -0,0 +1,24 @@
+package com.wire.kalium.logic.data.message.receipt
+
+import com.wire.kalium.logic.data.id.QualifiedID
+import com.wire.kalium.logic.data.user.ConnectionState
+import com.wire.kalium.logic.data.user.UserAvailabilityStatus
+import com.wire.kalium.logic.data.user.type.UserType
+import kotlinx.datetime.Instant
+
+data class DetailedReceipt(
+    val type: ReceiptType,
+    val date: Instant,
+    val userId: QualifiedID,
+    val userName: String?,
+    val userHandle: String?,
+    val userPreviewAssetId: QualifiedID?,
+    val userType: UserType,
+    val isUserDeleted: Boolean,
+    val connectionStatus: ConnectionState,
+    val availabilityStatus: UserAvailabilityStatus
```

Comment:
I think this segment of data is shareable with `MessageReaction`
We could use the same model for both.

```kt
data class UserSummmary(
    val userId: QualifiedID,
    val userName: String?,
    val userHandle: String?,
    val userPreviewAssetId: QualifiedID?,
    val userType: UserType,
    val isUserDeleted: Boolean,
    val connectionStatus: ConnectionState,
    val availabilityStatus: UserAvailabilityStatus
)


data class DetailedReceipt(
    val type: ReceiptType,
    val date: Instant,
    val userSummary: UserSummary
)

data class MessageReaction(
    val emoji: String,
    val userSummary: UserSummary
)
```